api/clients/v2/coretypes/eigenda_cert.go

```diff
@@ -4,34 +4,206 @@ import (
+		NonSignerStakesAndSignature: *nonSignerStakesAndSignature,
+		SignedQuorumNumbers:         quorumNumbers,
+	}, nil
+}
+
+// RelayKeys returns the relay keys used for reading blob contents from disperser relays
+func (c *EigenDACertV3) RelayKeys() []v2.RelayKey {
+	return c.BlobInclusionInfo.BlobCertificate.RelayKeys
+}
+
+// QuorumNumbers returns the quorum numbers requested
+func (c *EigenDACertV3) QuorumNumbers() []byte {
+	return c.BlobInclusionInfo.BlobCertificate.BlobHeader.QuorumNumbers
+}
+
+// RBN returns the reference block number
+func (c *EigenDACertV3) ReferenceBlockNumber() uint64 {
+	return uint64(c.BatchHeader.ReferenceBlockNumber)
+}
+
+// BlobVersion returns the blob version of the blob header
+func (c *EigenDACertV3) BlobVersion() v2.BlobVersion {
+	return c.BlobInclusionInfo.BlobCertificate.BlobHeader.Version
+}
+
+// ComputeBlobKey computes the blob key used for looking up the blob against an EigenDA network retrieval
+// entrypoint (e.g, a relay or a validator node)
+func (c *EigenDACertV3) ComputeBlobKey() (*v2.BlobKey, error) {
+	blobHeader := c.BlobInclusionInfo.BlobCertificate.BlobHeader
+	blobCommitments, err := c.Commitments()
+	if err != nil {
+		return nil, fmt.Errorf("blob commitments from protobuf: %w", err)
+	}
+
+	blobKeyBytes, err := v2.ComputeBlobKey(
+		blobHeader.Version,
+		*blobCommitments,
+		blobHeader.QuorumNumbers,
+		blobHeader.PaymentHeaderHash,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("compute blob key: %w", err)
+	}
+	blobKey, err := v2.BytesToBlobKey(blobKeyBytes[:])
+	if err != nil {
+		return nil, fmt.Errorf("bytes to blob key: %w", err)
+	}
+	return &blobKey, nil
+}
+
+func (c *EigenDACertV3) Serialize() ([]byte, error) {
+	certBytes, err := v3CertTypeEncodeArgs.Pack(c)
+	if err != nil {
+		return nil, fmt.Errorf("encode cert: %w", err)
+	}
+
+	return certBytes, nil
+}
+
+// Commitments returns the blob's cryptographic kzg commitments 
+func (c *EigenDACertV3) Commitments() (*encoding.BlobCommitments, error) {
+		// TODO: figure out how to remove this casting entirely
+		commitments := v2_cert_verifier.EigenDATypesV2BlobCommitment{
+			Commitment: v2_cert_verifier.BN254G1Point{
+				X: c.BlobInclusionInfo.BlobCertificate.BlobHeader.Commitment.Commitment.X,
+				Y: c.BlobInclusionInfo.BlobCertificate.BlobHeader.Commitment.Commitment.Y,
+			},
+			LengthCommitment: v2_cert_verifier.BN254G2Point{
+				X: c.BlobInclusionInfo.BlobCertificate.BlobHeader.Commitment.LengthCommitment.X,
+				Y: c.BlobInclusionInfo.BlobCertificate.BlobHeader.Commitment.LengthCommitment.Y,
+			},
+			LengthProof: v2_cert_verifier.BN254G2Point{
+				X: c.BlobInclusionInfo.BlobCertificate.BlobHeader.Commitment.LengthProof.X,
+				Y: c.BlobInclusionInfo.BlobCertificate.BlobHeader.Commitment.LengthProof.Y,
+			},
+			Length: c.BlobInclusionInfo.BlobCertificate.BlobHeader.Commitment.Length,
+		}
+	
+		blobCommitments, err := BlobCommitmentsBindingToInternal(&commitments)
```

Comment:
why do this conversion? shouldn't the EigenDACert() interface return the contract types instead of the eigenda backend types?

Reply:
this is because we have two bindings of the same thing - will follow up with a patch once patrick's PR is merged through

Reply:
was this fixed?